src/main/java/seedu/realodex/model/person/Birthday.java

```diff
@@ -35,22 +38,50 @@ public Birthday(String birthday) {
         this.birthday = birthdayDate;
     }
 
+    /**
+     * Constructs a default {@code Birthday}.
+     */
     public Birthday() {
-        this.birthday = Optional.empty();
+        SimpleDateFormat formatter = new SimpleDateFormat(INPUT_DATE_PATTERN, Locale.ENGLISH);
+        formatter.setLenient(false);
+        Optional<Date> birthdayDate;
+        try {
+            birthdayDate = Optional.of(formatter.parse("01May2023"));
+        } catch (ParseException e) {
+            birthdayDate = Optional.empty();
+        }
+        this.birthday = birthdayDate;
     }
 
     /**
      * Returns if a given string is a valid birthday.
      */
     public static boolean isValidBirthday(String birthday) {
-        if (birthday.isBlank() || birthday.equals("")) {
+        if (birthday.isBlank() || birthday.isEmpty()) {
             return true;
         }
         try {
             SimpleDateFormat formatter = new SimpleDateFormat(INPUT_DATE_PATTERN, Locale.ENGLISH);
             formatter.setLenient(false);
-            formatter.parse(birthday.trim());
-            return true;
+            Date parsedDate = formatter.parse(birthday.trim());
+
+            // Validate the month (1-12)
+            Calendar cal = Calendar.getInstance();
+            cal.setTime(parsedDate);
+            int month = cal.get(Calendar.MONTH);
+            if (month >= Calendar.JANUARY && month <= Calendar.DECEMBER) {
+                // Validate the day (1-31)
+                int day = cal.get(Calendar.DAY_OF_MONTH);
+                if (day >= 1 && day <= 31) {
+                    // Validate the year (e.g., not in the future)
```

Comment:
Good catch of future dates being invalid!

Reply:
Idk sia the check is not enforcing the YYYY i could type in 01May2 and its correct it sees as year '2'

Reply:
Nvm i will just tell them year >= 1000 is needed

Reply:
Good catch and good resolve!